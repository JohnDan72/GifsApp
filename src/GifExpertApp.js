import React, { useState } from 'react';
// import PropTypes from 'prop-types';
import {
    Carousel,
    Col,
    Container,
    Content,
    CustomProvider,
    Divider,
    FlexboxGrid,
    Header,
} from 'rsuite';
// icons
import { BsSearch } from 'react-icons/bs';

// components
import AddCategory from './components/AddCategory';
import NavBarHeader from './components/NavBarHeader';
import GifGrid from './components/GifGrid';



const GifExpertApp = ({defaultCats = []}) => {
    const [categories, setCategories] = useState(defaultCats);
    const [theme, settheme] = useState("dark");

    return (
        <CustomProvider theme={theme}>
            <Container >
                <Header>
                    <NavBarHeader settheme={settheme}/>
                </Header>
                <Content className="mt-1">
                    <FlexboxGrid justify="center">
                        <FlexboxGrid.Item as={Col} xs={24}>
                            <Carousel    
                                        className="custom-slider"
                                        shape='bar'
                                        >
                                <div id="carousel1">
                                    <div className="v-a-middle" justify="center">
                                        <h3>Gif Expert</h3>
                                    </div>
                                </div>
                                <div id="carousel2">
                                    <div className="v-a-middle" justify="center">
                                        <h3>Busca gifs de tu interés</h3>
                                    </div>
                                </div>
                                <div id="carousel3">
                                    <div className="v-a-middle" justify="center">
                                        <h3>Contamos con más de 1,000,000 de posibilidades</h3>
                                    </div>
                                </div>
                                <div id="carousel4">
                                    <div className="v-a-middle" justify="center">
                                        <h3>Encuentra solo lo mejor</h3>
                                    </div>
                                </div>
                            </Carousel>
                        </FlexboxGrid.Item>
                        <FlexboxGrid.Item as={Col} colspan={24} xs={20} className="mt-5">
                            <h3>Gif Expert App</h3>
                            <Divider>
                                <BsSearch style={{ fontSize: '1.2em' }}/>
                            </Divider>
                            <FlexboxGrid justify="center">
                                <FlexboxGrid.Item as={Col} xs={12}>
                                    <AddCategory categories={categories} setCategories={setCategories}/>

                                </FlexboxGrid.Item>

                            </FlexboxGrid>

                            <Content className="my-5 ">
                            {
                                categories.map((cat) => (
                                    <GifGrid category={cat} key={cat}/>
                                ))
                            }

                            </Content>
                            
                            
                        </FlexboxGrid.Item>
                    </FlexboxGrid>
                </Content>
            </Container>

        </ CustomProvider>
    );
}

export default GifExpertApp;