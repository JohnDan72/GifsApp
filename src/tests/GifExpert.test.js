import React from "react";
import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp";


describe('Pruebas Gif Expert' , () => {
    
    let wrapper = shallow( <GifExpertApp /> );

    beforeEach(() => {
        wrapper = shallow(<GifExpertApp />);
    })

    test('should render GifExpert ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should show an categories list', () => {
        const categories = ['one punch','dragon ball'];
        const wrapper = shallow( <GifExpertApp defaultCats={categories} />);

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('GifGrid').length).toBe(categories.length)

    })
    
})