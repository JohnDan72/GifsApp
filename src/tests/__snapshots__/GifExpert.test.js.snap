// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Gif Expert should render GifExpert  1`] = `
<CustomProvider
  theme="dark"
>
  <Container>
    <Header>
      <NavBarHeader
        settheme={[Function]}
      />
    </Header>
    <Content
      className="mt-1"
    >
      <FlexboxGrid
        justify="center"
      >
        <FlexboxGridItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "as": [Function],
                "className": [Function],
                "classPrefix": [Function],
                "lg": [Function],
                "lgHidden": [Function],
                "lgOffset": [Function],
                "lgPull": [Function],
                "lgPush": [Function],
                "md": [Function],
                "mdHidden": [Function],
                "mdOffset": [Function],
                "mdPull": [Function],
                "mdPush": [Function],
                "sm": [Function],
                "smHidden": [Function],
                "smOffset": [Function],
                "smPull": [Function],
                "smPush": [Function],
                "xs": [Function],
                "xsHidden": [Function],
                "xsOffset": [Function],
                "xsPull": [Function],
                "xsPush": [Function],
              },
              "render": [Function],
            }
          }
          xs={24}
        >
          <Carousel
            className="custom-slider"
            shape="bar"
          >
            <div
              id="carousel1"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Gif Expert
                </h3>
              </div>
            </div>
            <div
              id="carousel2"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Busca gifs de tu interés
                </h3>
              </div>
            </div>
            <div
              id="carousel3"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Contamos con más de 1,000,000 de posibilidades
                </h3>
              </div>
            </div>
            <div
              id="carousel4"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Encuentra solo lo mejor
                </h3>
              </div>
            </div>
          </Carousel>
        </FlexboxGridItem>
        <FlexboxGridItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "as": [Function],
                "className": [Function],
                "classPrefix": [Function],
                "lg": [Function],
                "lgHidden": [Function],
                "lgOffset": [Function],
                "lgPull": [Function],
                "lgPush": [Function],
                "md": [Function],
                "mdHidden": [Function],
                "mdOffset": [Function],
                "mdPull": [Function],
                "mdPush": [Function],
                "sm": [Function],
                "smHidden": [Function],
                "smOffset": [Function],
                "smPull": [Function],
                "smPush": [Function],
                "xs": [Function],
                "xsHidden": [Function],
                "xsOffset": [Function],
                "xsPull": [Function],
                "xsPush": [Function],
              },
              "render": [Function],
            }
          }
          className="mt-5"
          colspan={24}
          xs={20}
        >
          <h3>
            Gif Expert App
          </h3>
          <Divider>
            <BsSearch
              style={
                Object {
                  "fontSize": "1.2em",
                }
              }
            />
          </Divider>
          <FlexboxGrid
            justify="center"
          >
            <FlexboxGridItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "as": [Function],
                    "className": [Function],
                    "classPrefix": [Function],
                    "lg": [Function],
                    "lgHidden": [Function],
                    "lgOffset": [Function],
                    "lgPull": [Function],
                    "lgPush": [Function],
                    "md": [Function],
                    "mdHidden": [Function],
                    "mdOffset": [Function],
                    "mdPull": [Function],
                    "mdPush": [Function],
                    "sm": [Function],
                    "smHidden": [Function],
                    "smOffset": [Function],
                    "smPull": [Function],
                    "smPush": [Function],
                    "xs": [Function],
                    "xsHidden": [Function],
                    "xsOffset": [Function],
                    "xsPull": [Function],
                    "xsPush": [Function],
                  },
                  "render": [Function],
                }
              }
              xs={12}
            >
              <AddCategory
                categories={Array []}
                setCategories={[Function]}
              />
            </FlexboxGridItem>
          </FlexboxGrid>
          <Content
            className="my-5 "
          />
        </FlexboxGridItem>
      </FlexboxGrid>
    </Content>
  </Container>
</CustomProvider>
`;

exports[`Pruebas Gif Expert should show an categories list 1`] = `
<CustomProvider
  theme="dark"
>
  <Container>
    <Header>
      <NavBarHeader
        settheme={[Function]}
      />
    </Header>
    <Content
      className="mt-1"
    >
      <FlexboxGrid
        justify="center"
      >
        <FlexboxGridItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "as": [Function],
                "className": [Function],
                "classPrefix": [Function],
                "lg": [Function],
                "lgHidden": [Function],
                "lgOffset": [Function],
                "lgPull": [Function],
                "lgPush": [Function],
                "md": [Function],
                "mdHidden": [Function],
                "mdOffset": [Function],
                "mdPull": [Function],
                "mdPush": [Function],
                "sm": [Function],
                "smHidden": [Function],
                "smOffset": [Function],
                "smPull": [Function],
                "smPush": [Function],
                "xs": [Function],
                "xsHidden": [Function],
                "xsOffset": [Function],
                "xsPull": [Function],
                "xsPush": [Function],
              },
              "render": [Function],
            }
          }
          xs={24}
        >
          <Carousel
            className="custom-slider"
            shape="bar"
          >
            <div
              id="carousel1"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Gif Expert
                </h3>
              </div>
            </div>
            <div
              id="carousel2"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Busca gifs de tu interés
                </h3>
              </div>
            </div>
            <div
              id="carousel3"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Contamos con más de 1,000,000 de posibilidades
                </h3>
              </div>
            </div>
            <div
              id="carousel4"
            >
              <div
                className="v-a-middle"
                justify="center"
              >
                <h3>
                  Encuentra solo lo mejor
                </h3>
              </div>
            </div>
          </Carousel>
        </FlexboxGridItem>
        <FlexboxGridItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "as": [Function],
                "className": [Function],
                "classPrefix": [Function],
                "lg": [Function],
                "lgHidden": [Function],
                "lgOffset": [Function],
                "lgPull": [Function],
                "lgPush": [Function],
                "md": [Function],
                "mdHidden": [Function],
                "mdOffset": [Function],
                "mdPull": [Function],
                "mdPush": [Function],
                "sm": [Function],
                "smHidden": [Function],
                "smOffset": [Function],
                "smPull": [Function],
                "smPush": [Function],
                "xs": [Function],
                "xsHidden": [Function],
                "xsOffset": [Function],
                "xsPull": [Function],
                "xsPush": [Function],
              },
              "render": [Function],
            }
          }
          className="mt-5"
          colspan={24}
          xs={20}
        >
          <h3>
            Gif Expert App
          </h3>
          <Divider>
            <BsSearch
              style={
                Object {
                  "fontSize": "1.2em",
                }
              }
            />
          </Divider>
          <FlexboxGrid
            justify="center"
          >
            <FlexboxGridItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "as": [Function],
                    "className": [Function],
                    "classPrefix": [Function],
                    "lg": [Function],
                    "lgHidden": [Function],
                    "lgOffset": [Function],
                    "lgPull": [Function],
                    "lgPush": [Function],
                    "md": [Function],
                    "mdHidden": [Function],
                    "mdOffset": [Function],
                    "mdPull": [Function],
                    "mdPush": [Function],
                    "sm": [Function],
                    "smHidden": [Function],
                    "smOffset": [Function],
                    "smPull": [Function],
                    "smPush": [Function],
                    "xs": [Function],
                    "xsHidden": [Function],
                    "xsOffset": [Function],
                    "xsPull": [Function],
                    "xsPush": [Function],
                  },
                  "render": [Function],
                }
              }
              xs={12}
            >
              <AddCategory
                categories={
                  Array [
                    "one punch",
                    "dragon ball",
                  ]
                }
                setCategories={[Function]}
              />
            </FlexboxGridItem>
          </FlexboxGrid>
          <Content
            className="my-5 "
          >
            <GifGrid
              category="one punch"
              key="one punch"
            />
            <GifGrid
              category="dragon ball"
              key="dragon ball"
            />
          </Content>
        </FlexboxGridItem>
      </FlexboxGrid>
    </Content>
  </Container>
</CustomProvider>
`;
